### PR TITLE
Data model

### PR DESCRIPTION
**Pull request by christianbiring1**
In this project, I had to use migration files to build the schema.

### CODE REVIEW

File: db/migrate/20220820130343_add_user_ref_to_comments.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
+  end
+end
```
Comment (medaminedev66) [inline db/migrate/20220820130343_add_user_ref_to_comments.rb]: - [ ] The reference is not directly to user, it is instead author to table users. Kindly see the below example on how to make such migration for adding a reference with a different name than the reference model class name and make changes in your database. 🙏

image

File: db/migrate/20220820131321_add_user_ref_to_likes.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
+  end
+end
```
Comment (medaminedev66) [inline db/migrate/20220820131321_add_user_ref_to_likes.rb]: - [ ] Again, the reference name should be `author`. Please refer to the above similar comment for the other migration.

File: db/migrate/20220820131829_add_user_ref_to_post.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPost < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
+  end
+end
```
Comment (medaminedev66) [inline db/migrate/20220820131829_add_user_ref_to_post.rb]: - [ ] Again, the reference name should be `author`. Please refer to the above similar comment for the other migration.

File: db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  add_foreign_key "comments", "posts"
+  add_foreign_key "comments", "users"
+  add_foreign_key "likes", "posts"
+  add_foreign_key "likes", "users"
+  add_foreign_key "posts", "users"
+end
```
Comment (medaminedev66) [inline db/schema.rb]: After applying the requested changes in the migrations, this should look like this:

image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @christianbiring1  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional
- There are no linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 5]
```diff
@@ -1,24 +1,51 @@
 # README
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+# Math_magician
```
Comment (medaminedev66) [inline README.md]: - [ ] Please update the name of the project in your Readme.

File: README.md [hunk 6]
```diff
@@ -1,24 +1,51 @@
...
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Hat tip to anyone who will take a glance to this project
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (medaminedev66) [inline README.md:50]: [OPTIONAL] This MIT link is broken. I recommend fixing this. Your README is the entry point of your users and having it well written without any broken links gives a good first impression and maximizes the potential of your product to your users.

File: db/migrate/20220820130343_add_user_ref_to_comments.rb (hunk 1 above)
Comment (christianbiring1) [reply db/migrate/20220820130343_add_user_ref_to_comments.rb]: Alright.
Thank you @medaminedev66 for the hint.

File: db/migrate/20220820131321_add_user_ref_to_likes.rb (hunk 2 above)
Comment (christianbiring1) [reply db/migrate/20220820131321_add_user_ref_to_likes.rb]: Yeah!, :+1:

File: db/migrate/20220820131829_add_user_ref_to_post.rb (hunk 3 above)
Comment (christianbiring1) [reply db/migrate/20220820131829_add_user_ref_to_post.rb]: Got it :+1:

File: db/schema.rb (hunk 4 above)
Comment (christianbiring1) [reply db/schema.rb]: Alright!. I got it. :+1:

File: db/migrate/20220822114132_create_posts.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :tite
```
Comment (rloterh) [inline db/migrate/20220822114132_create_posts.rb]: - [ ] OPTIONAL
I'll recommend you correct the field name on line 4 from `tite` to `title`

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Christian,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉🌟 

## Highlights 👍
- Migration follows given ERD

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comments marked **[OPTIONAL]** under this review_
<br>
Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220822114132_create_posts.rb (hunk 7 above)
Comment (christianbiring1) [reply db/migrate/20220822114132_create_posts.rb]: Yeah, sure! Thank you @rloterh :rocket:

**Pull request merged**